extension/data_loader/mmap_data_loader.cpp

```diff
@@ -182,12 +183,24 @@ Result<FreeableBuffer> MmapDataLoader::load(
   Range range =
       get_overlapping_pages(static_cast<uintptr_t>(offset), size, page_size_);
 
+  size_t map_size = range.size;
+#ifdef _WIN32
+  // On Windows, don't mmap-in memory past end of on-disk file.
+  //
+  // The Windows implementation of mmap uses CreateFileMapping which returns
+  // error STATUS_SECTION_TOO_BIG (0xc0000040) if we try to map past the end
+  // of the last page of a file mapped in as read-only.
+  if (range.start + range.size > file_size_) {
+    map_size = file_size_ - range.start;
+  }
+#endif
```

Comment:
Seems reasonable to always do this, if it works on unix systems; we're trying to minimize the number of ifdefs in common code.
```suggestion
  size_t map_size = range.size;
  if (range.start + map_size > file_size_) {
    // Clamp to the end of the file.
    //
    // The Windows implementation of mmap uses CreateFileMapping which returns
    // error STATUS_SECTION_TOO_BIG (0xc0000040) if we try to map past the end
    // of the last page of a file mapped in as read-only.
    map_size = file_size_ - range.start;
  }
```